### ISSUE TITLE
`cargo kani` failure (`std` / `core` contents not found)

### ISSUE THREAD

**Issue by wbrickner** [model-checking/kani#1180]
using the following command line invocation:

```
cargo kani
```

with Kani version: `cargo-kani 0.1.0`

I expected to see: the build succeed.

Instead, this happened:

```
...

error[E0531]: cannot find tuple struct or tuple variant `Ok` in this scope
   --> /Users/wbrickner/.cargo/registry/src/github.com-1ecc6299db9ec823/paste-1.0.7/src/lib.rs:411:9
    |
411 |         Ok(ident) => TokenTree::Ident(ident),
    |         ^^ not found in this scope

error[E0531]: cannot find tuple struct or tuple variant `Err` in this scope
   --> /Users/wbrickner/.cargo/registry/src/github.com-1ecc6299db9ec823/paste-1.0.7/src/lib.rs:412:9
    |
412 |         Err(_) => {
    |         ^^^ not found in this scope

error[E0425]: cannot find function, tuple struct or tuple variant `Err` in this scope
   --> /Users/wbrickner/.cargo/registry/src/github.com-1ecc6299db9ec823/paste-1.0.7/src/lib.rs:413:20
    |
413 |             return Err(Error::new(
    |                    ^^^ not found in this scope

error[E0425]: cannot find function, tuple struct or tuple variant `Ok` in this scope
   --> /Users/wbrickner/.cargo/registry/src/github.com-1ecc6299db9ec823/paste-1.0.7/src/lib.rs:421:5
    |
421 |     Ok(tokens)
    |     ^^ not found in this scope

error: requires `sized` lang_item

error: requires `receiver` lang_item

Some errors have detailed explanations: E0412, E0425, E0433, E0463, E0531.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `paste` due to 220 previous errors
Error: cargo exited with status exit status: 101

```

Seems like `rustc` is pretending it doesn't even know about the `core` constructs like `Ok` / `Err`.
I have no idea how this is happening or why, looking for help.

Interestingly, if I do not use `cargo kani` (`kani`), then I am able to formally verify as I expect.

Sorry if this is too little information.  Clean install from today.

**Comment by zhassan-aws** [comment]
Hi @wbrickner. Thanks for reporting this issue. This sounds very similar to #1178. Did your package use specific crates as dependencies?

**Comment by zhassan-aws** [comment]
From the output, it looks like [paste](https://crates.io/crates/paste) is used as a dependency? If so, then this is the same as #1178.

**Comment by wbrickner** [comment]
Yes, I depend on my own proc macro crate locally as well as `parse`. 

However, I've removed all dependencies and tried a single proof function like

```rust
#[cfg(kani)]
#[kani::proof]
fn trivial() {
  let x: f64 = kani::any();
  assert!(f64::is_finite(x));  // will fail (NaN, ±Infinity)
}
```

which results in the same behavior from `cargo kani`.

The other commonality between my proofs and that of `rust-decimal` is we're both probably using `f64` as `kani::any()`, but... I don't think that's the issue.

**Comment by zhassan-aws** [comment]
The following worked for me:
1. `cargo new --lib f64_check`
2. `cd f64_check`
3. Pasted the example you posted in `src/lib.rs`
4. Run `cargo kani`.

This is the output I got:
```

RESULTS:
Check 1: trivial.assertion.1
         - Status: FAILURE
         - Description: "assertion failed: f64::is_finite(x)"
         - Location: src/lib.rs:5:3 in function trivial

Check 2: <T as kani::Arbitrary>::any.unsupported_construct.1
         - Status: SUCCESS
         - Description: "resume instruction"
         - Location: ~/.kani/kani-0.1.0/library/kani/src/arbitrary.rs:17:5 in function <T as kani::Arbitrary>::any


SUMMARY:
 ** 1 of 2 failed
Failed Checks: assertion failed: f64::is_finite(x)
 File: "/home/zhassan/rust/f64_check/src/lib.rs", line 5, in trivial
```
Can you try the above steps?

**Comment by wbrickner** [comment]
Ok, I guess I was mistaken before in my report.

- Repeating your process works.
- Reintroducing either `paste` or `my_custom_proc_macro` causes the same build failure.

This must be something to do with the handling of proc macro crates / dependencies?

**Comment by zhassan-aws** [comment]
In #1178, the issue is due to the dependent crates using a `build.rs` script, which Kani is not handling correctly.

**Comment by zhassan-aws** [comment]
This issue was fixed in the 0.2 release, so closing it. Feel free to reopen if it's still occurring for you.

https://github.com/model-checking/kani/releases/tag/kani-0.2.0
